lib/app/view/system_tray.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:io';
```

Comment:
I am not sure if I used the correct labels, but this should work, also you dont need to make the list of items late, just create them in the function. But also check if the context.mounted
here is the compelte file so you can easily copy paste

```dart
import 'dart:io';

import 'package:flutter/material.dart';
import 'package:tray_manager/tray_manager.dart';
import 'package:window_manager/window_manager.dart';

import '../../l10n/l10n.dart';

String trayIcon() {
  if (Platform.isWindows) {
    return 'assets/images/tray_icon.ico';
  } else {
    return 'assets/images/tray_icon.png';
  }
}

class SystemTray with TrayListener {
  Future<void> init() async {
    trayManager.addListener(this);
    await trayManager.setIcon(trayIcon());
  }

  Future<void> dispose() async {
    trayManager.removeListener(this);
  }

  Future<void> updateTrayMenuItems(
    BuildContext context,
  ) async {
    bool isVisible = await windowManager.isVisible();

    if (context.mounted) {
      final trayMenuItems = [
        MenuItem(
          key: 'show_hide_window',
          label: isVisible ? context.l10n.hideToTray : context.l10n.closeApp,
        ),
        MenuItem.separator(),
        MenuItem(
          key: 'close_application',
          label: context.l10n.closeApp,
        ),
      ];

      await trayManager.setContextMenu(Menu(items: trayMenuItems));
    }
  }

  @override
  void onTrayMenuItemClick(MenuItem menuItem) {
    switch (menuItem.key) {
      case 'show_hide_window':
        windowManager.isVisible().then((value) {
          if (value) {
            windowManager.hide();
          } else {
            windowManager.show();
          }
        });
      case 'close_application':
        windowManager.close();
    }
  }
}
```